crates/task-impls/src/transactions.rs

```diff
@@ -219,30 +219,15 @@ where
                 // TODO (Keyao) Determine whether to allow empty blocks.
                 // <https://github.com/EspressoSystems/HotShot/issues/1822>
                 let txns = self.wait_for_transactions(parent_leaf).await?;
+                let encoded_txns = VIDTransaction::encode(txns.clone());
```

Comment:
We need to use the application-specific encoding function here, ie `let (payload, payload_metadata) = TYPES::BlockPayload::build(txns); let encoded_txns = block.encode();`. This also means we can restrict to using `VIDTransaction` anymore, we have to deal with the application-specific transaction type (in the sequencer, the transaction type will have an extra namespace field, which we will use to populate the namespace table in `BockPayload::build`.